salt/_modules/metalk8s.py

```diff
@@ -118,6 +102,39 @@ def add_etcd_node(host, endpoint=None):
         raise exn
 
 
+@depends('KUBERNETES_PRESENT', fallback_function=deps_missing)
+def add_etcd_node(host, endpoint=None):
+    '''Add a new `etcd` node into the `etcd` cluster.
+
+    This module is only runnable from the salt-master on the bootstrap node.
+
+    Arguments:
+        host (str): hostname of the new etcd node
+        endpoint (str): endpoint of the new etcd node
+    '''
+    if not endpoint:
+        # If we have no endpoint get it from mine
+        node_ip = __salt__['saltutil.runner'](
```

Comment:
This is a `module`, so we shouldn't rely on runners IMHO.

Reply:
Yes but if we call the classic mine.update then if we use this module on the master directly pretty sure that doesn't work. 
And also even if we run on a minion `salt-call saltuil.runner mine.get` works as exepected

Reply:
And btw this is only a workaround waiting for #877